From 05ae2fd020de1c329735bba9a53d2359ad430283 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Fri, 19 Oct 2018 17:45:50 +0300
Subject: [PATCH 125/741] staging: fsl-dpaa2/mac: Fix dpmac_set_link_state()
 command

commit f9ec73255ddb11432106b656fb50361ccd5a9b25 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The instruction writing link state value in the MC command
structure wasn't correct, but it happened to work nonetheless.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/dpmac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/dpmac.c b/drivers/staging/fsl-dpaa2/mac/dpmac.c
index 9ca830dd2131..59606d1d23dd 100644
--- a/drivers/staging/fsl-dpaa2/mac/dpmac.c
+++ b/drivers/staging/fsl-dpaa2/mac/dpmac.c
@@ -515,7 +515,7 @@ int dpmac_set_link_state(struct fsl_mc_io *mc_io,
 	cmd_params = (struct dpmac_cmd_set_link_state *)cmd.params;
 	cmd_params->options = cpu_to_le64(link_state->options);
 	cmd_params->rate = cpu_to_le32(link_state->rate);
-	cmd_params->up = dpmac_get_field(link_state->up, STATE);
+	dpmac_set_field(cmd_params->up, STATE, link_state->up);
 
 	/* send command to mc*/
 	return mc_send_command(mc_io, &cmd);
-- 
2.17.1

